crates/libcontainer/src/process/fork.rs

```diff
@@ -8,15 +8,16 @@ use nix::unistd::Pid;
 // using clone, we would have to manually make sure all the variables are
 // correctly send to the new process, especially Rust borrow checker will be a
 // lot of hassel to deal with every details.
-pub fn container_fork<F: FnOnce() -> Result<()>>(cb: F) -> Result<Pid> {
+pub fn container_fork<F: FnOnce() -> Result<i32>>(cb: F) -> Result<Pid> {
```

Comment:
Nit: I would suggest adding a bit of comment explaining the requirement for returning an error code. The `i32` unfortunately will take some time to decipher if people are not familiar with the code.

Reply:
can you take a look at this https://github.com/utam0k/youki/pull/115/files#diff-b0f79d7a9d35c79d1dfa4d453d78572414d768506564d63a2504d723e69b43b5R12-R14